### PR TITLE
Stig

### PR DESCRIPTION
**Pull request by StigNorland**

### CODE REVIEW

File: src/main/java/no/sikt/sws/WorkspaceIndexHandler.java [hunk 1]
```diff
@@ -34,15 +34,19 @@ protected IndexResponse processInput(String body, RequestInfo request, Context c
...
             var url = workspace + "-" + index;
             logger.info("URL: " + url);
-            var response = openSearchClient.sendRequest(httpMethod, url, body);
+
+            var response = openSearchClient.sendRequest(
+                    httpMethod,
+                    url,
+                    RequestUtil.addWorkspace(body, workspace, index));
             logger.info("response-code:" + response.getStatus());
-            logger.info("response-body:" + response.getBody());
```
Comment (sondrev) [inline src/main/java/no/sikt/sws/WorkspaceIndexHandler.java:39]: Can you readd this line? Its usefull for debugging right now to check what the output is before you strip it

File: src/main/java/no/sikt/sws/WorkspaceIndexHandler.java [hunk 2]
```diff
@@ -34,15 +34,19 @@ protected IndexResponse processInput(String body, RequestInfo request, Context c
         try {
             var url = workspace + "-" + index;
             logger.info("URL: " + url);
-            var response = openSearchClient.sendRequest(httpMethod, url, body);
+
+            var response = openSearchClient.sendRequest(
+                    httpMethod,
+                    url,
+                    RequestUtil.addWorkspace(body, workspace, index));
```
Comment (sondrev) [inline src/main/java/no/sikt/sws/WorkspaceIndexHandler.java:41]: Is this needed? Is there a usercase in which you put a new index with a config, and that config has the workspace in it?

File: src/main/java/no/sikt/sws/RequestUtil.java [hunk 3]
```diff
@@ -28,4 +30,16 @@ public static String getWorkspace(RequestInfo request) {
...
     }
+
+    public static String addWorkspace(String body, String workspace, String index) {
+        if (body == null) {
+            return null;
+        }
+
+        return attempt(() -> {
+            var strippedIndex = Arrays.stream(index.split("/")).findFirst();
+            return body.replaceAll(index, workspace + "-" + strippedIndex);
```
Comment (sondrev) [inline src/main/java/no/sikt/sws/RequestUtil.java:41]: I can see alot of issues with this if the index is named somethigng short..  you should see if there is a better way of doing it. Like figuring out the excact usecases (json-bodies)

File: docs/http-client.env.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
+  "dev": {
+    "workspace": "sondre"
```
Comment (sondrev) [inline docs/http-client.env.json:3]: whats this file used for?

**Review by sondrev — CHANGES_REQUESTED** [review]
Please add some tests for this

File: docs/http-client.env.json (hunk 4 above)
Comment (StigNorland) [reply docs/http-client.env.json:3]: I wasn't aware that it came from me, it is from intellij and can be removed.

File: src/main/java/no/sikt/sws/WorkspaceIndexHandler.java (hunk 2 above)
Comment (StigNorland) [reply src/main/java/no/sikt/sws/WorkspaceIndexHandler.java]: "workspaces" are a concept that we implement to hide other indexes from each client, this code translate between gateway and opensearch.

File: src/main/java/no/sikt/sws/WorkspaceIndexHandler.java (hunk 1 above)
Comment (StigNorland) [reply src/main/java/no/sikt/sws/WorkspaceIndexHandler.java:38]: You can do whatever you want with it, read output before stripping or/and after. I really don't have an opinion about it, put it where ever you want it.

File: src/main/java/no/sikt/sws/WorkspaceIndexHandler.java (hunk 2 above)
Comment (StigNorland) [reply src/main/java/no/sikt/sws/WorkspaceIndexHandler.java]: However, I think this should be moved to the client, to ensure that all request uses the same informationflow.
If you have an idea about a more descriptive name, please enlighten me.

**Pull request closed without merging**